client/python/gradio_client/compatibility.py

```diff
@@ -61,8 +61,6 @@ def _inner(*data):
             if not self.is_valid:
                 raise utils.InvalidAPIEndpointError()
             data = self.insert_state(*data)
-            if self.client.upload_files:
```

Comment:
Maybe we should still `serialize` the data here. I think this is breaking backwards compatibility with 3.x